app/src/main/java/com/example/marvelapp/api/service/CharacterServiceImpl.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.marvelapp.api.service
+
+import com.example.marvelapp.api.MarvelAppApi
+import com.example.marvelapp.api.mapper.CharacterMapper
+import com.example.marvelapp.api.util.Result
+import com.example.marvelapp.entity.Character
+
+class CharacterServiceImpl : CharacterService {
+
+    private val api = ServiceGenerator()
+    private val mapper = CharacterMapper()
+
+    override fun getCharacters(): Result<List<Character>> {
+        try {
+            val callResponse = api.createService(MarvelAppApi::class.java).getCharacters()
+            val response = callResponse.execute()
+            if (response.isSuccessful)
+                response.body()?.let {
+                    mapper.transform(it)
+                }?.let {
+                    return Result.Success(it)
+                }
+        } catch (e: Exception) {
+            return Result.Failure(e)
+        }
+        return Result.Failure(Exception(NOT_FOUND))
+    }
+
+    companion object {
+        private const val NOT_FOUND = "Characters not found"
```

Comment:
If you will show this text on the screen, this should be in a string resource, never hardcoded here

Reply:
done